LightBlue/Standalone/StandaloneList.cs

```diff
@@ -1,37 +1,26 @@
+using Azure.Storage.Blobs.Models;
 using System;
+using System.Collections.Generic;
 using System.Globalization;
 using System.IO;
 using System.Linq;
 
-using Microsoft.WindowsAzure.Storage.Blob;
 
 namespace LightBlue.Standalone
 {
     internal static class StandaloneList
     {
-        public static IAzureBlobResultSegment ListBlobsSegmentedAsync(
+        public static List<StandaloneAzureBlockBlob> ListBlobsSegmentedAsync(
             string containerDirectory,
-            string searchDirectory,
             string prefix,
-            BlobListing blobListing,
-            BlobListingDetails blobListingDetails,
-            int? maxResults,
-            BlobContinuationToken currentToken)
+            BlobTraits blobTraits,
+            int maxResults)
         {
-            if (blobListing == BlobListing.Hierarchical && (blobListingDetails & BlobListingDetails.Snapshots) == BlobListingDetails.Snapshots)
-            {
-                throw new ArgumentException("Listing snapshots is only supported in flat mode.");
-            }
-
-            var numberToSkip = DetermineNumberToSkip(currentToken);
-
-            var resultSegment = blobListing == BlobListing.Flat
-                ? FindFilesFlattened(containerDirectory, searchDirectory, prefix, maxResults, numberToSkip)
-                : FindFilesHierarchical(containerDirectory, searchDirectory, prefix, maxResults, numberToSkip);
+            var resultSegment = FindFilesFlattened(containerDirectory, prefix, maxResults);
```

Comment:
I assume 'Listing Snapshots' and 'Hierarchical Mode' are not used and are now unsupported?

Reply:
As well as 'Continuation' and 'Skipping'?

Reply:
Hierachical listing was not used in our code bases so I've deprecated support for the feature.

Snapshot listing should still be possible by passing `BlobStates.Snapshots`

The underlying API has adopted AsyncEnumerable to support continuation. Our code bases were always performing full enumerations so I took the lazy approach and just implemented full enumeration.